filecopy/operations/services/notification/models.py

```diff
@@ -0,0 +1,41 @@
+# Copyright (C) 2022 Indoc Research
+#
+# This program is free software: you can redistribute it and/or modify it under the terms of the GNU Affero General
+# Public License as published by the Free Software Foundation, either version 3 of the License, or any later version.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+# without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License along with this program.
+# If not, see http://www.gnu.org/licenses/.
+
+from typing import List
+from typing import Optional
+
+from pydantic import BaseModel
+
+
+class Location(BaseModel):
+    id: str
+    path: str
+    zone: int
+
+
+class Target(BaseModel):
+    id: str
+    name: str
+    type: Optional[str]
+
+
+class PipelineNotification(BaseModel):
+    type: str
+    recipient_username: str
+    involved_as: str
```

Comment:
Please use here `InvolvementType` enum as in documentation.
```
class InvolvementType(str, Enum):
    INITIATOR = 'initiator'
    OWNER = 'owner'
    RECEIVER = 'receiver'
```

```suggestion
    involved_as: InvolvementType
```